### ISSUE TITLE
NullReferenceException thrown when the start of the text is about to become `begine`, `process` or `end`

### ISSUE THREAD

**Issue by ghost** [PowerShell/PSReadLine#1079]
![2019-09-29 (4)](https://user-images.githubusercontent.com/44871748/65841243-3ebd1000-e2d5-11e9-9693-f69e3dc28d48.png)

Environment data
----------------

<!-- provide the output of the following:
```powershell
& {
    "PS version: $($PSVersionTable.PSVersion)"
    $v = (Get-Module PSReadline).Version
    $m = Get-Content "$(Split-Path -Parent (Get-Module PSReadLine).Path)\PSReadLine.psd1" | Select-String "Prerelease = '(.*)'"
    if ($m) {
        $v = "$v-" + $m.Matches[0].Groups[1].Value
    }
    "PSReadline version: $v"
    if ($IsLinux -or $IsMacOS) {
        "os: $(uname -a)"
    } else {
        "os: $((dir $env:SystemRoot\System32\cmd.exe).VersionInfo.FileVersion)"
    }
    "PS file version: $($name = if ($PSVersionTable.PSEdition -eq "Core") { "pwsh.dll" } else { "powershell.exe" }; (dir $pshome\$name).VersionInfo.FileVersion)"
    "BufferWidth: $([console]::BufferWidth)"
    "BufferHeight: $([console]::BufferHeight)"
}
```Use-WindowsUnattend -ScratchDirectory 0x4d644045
-->Read `em and weep...!

Steps to reproduce or exception report
--------------------------------------

**Comment by daxian-dbw** [comment]
Thanks for the reporting. This issue was fixed by #1049.

**Comment by daxian-dbw** [comment]
Rename the issue so it's easy for other people to search.
